### PR TITLE
feat: Add option to switch to rapidjson for deserializing JSON

### PR DESCRIPTION
**Pull request by lynnagara**
This is an experiment to see if we can improve performance of the post
process forwarder. The "post-process-forwarder:rapidjson" setting toggles
switching between the simplejson and rapidjson libraries for parsing the event
payload. The default value is false (i.e. not using rapidjson).

### CODE REVIEW

File: tests/sentry/eventstream/kafka/test_protocol.py [hunk 1]
```diff
@@ -12,11 +12,13 @@
 from sentry.utils import json
 
 
+@pytest.mark.django_db
```
Comment (untitaker) [inline tests/sentry/eventstream/kafka/test_protocol.py:15]: I think mocking options is better here

**Review by untitaker — APPROVED** [review]
neat.. have been meaning to try orjson

**Review by joshuarli — APPROVED** [review]
I have a feeling `orjson` is a more suitable choice longer term as it's more featureful (being able to serialize more things), but rapidjson is way more drop-in.

**Comment by lynnagara** [comment]
@joshuarli What kind of things are you expecting rapidjson won't be able to serialize? Afaik they all support arbitrary types in the same way with the default function.

**Comment by joshuarli** [comment]
> @joshuarli What kind of things are you expecting rapidjson won't be able to serialize? Afaik they all support arbitrary types in the same way with the default function.

Oh, you're right about that. orjson just claims better performance for orjson-natively-supported types, like dataclasses.

Some more good things about orjson - seems the ecosystem likes it, for example there's https://pypi.org/project/drf-orjson-renderer/ which I mention in https://github.com/getsentry/sentry/pull/25830 and structlog [recommends it](https://www.structlog.org/en/stable/performance.html).

**Pull request merged**